### PR TITLE
[FEAT] grant search ui / api implementation

### PR DESCRIPTION
**Pull request by UcDust**

### CODE REVIEW

**Comment by qjhart** [comment]
@UcDust , I left this sort of Broken, but just so you can see the move toward search/browse.  There are now two endpoints you can setup your browse endpoints `/api/expert/browse` and `/apo/grant/browse`.  The `/api/browse` returns both.  I added a `@type` into that for determining which is which.  I'll start looking at search the same way.

File: services/base-service/models/middleware.js [hunk 1]
```diff
@@ -293,6 +293,37 @@ const openapi = OpenAPI(
...
+            items: {
+              type: "string",
+              enum: ["collab","community","industry","media"]
+            }
+          },
+          style: "simple",
+          explode: false
+        },
+        type: {
+            in: "query",
```
Comment (qjhart) [inline services/base-service/models/middleware.js:313]: @UcDust this is the new search parameter

**Comment by qjhart** [comment]
@UcDust search has been implemented to return grants and experts now.

**Comment by UcDust** [comment]
Discussed with @qjhart in slack already, but to document here.., we'll need to support the following:

- [ ] filtering grants that are active and/or complete, so another param similar to `type` to further refine results
- [ ] return counts for each type of result, for the number of experts, grants, active vs completed etc, that shows up in the search filters (see image below)

image

**Comment by qjhart** [comment]
@UcDust / @rakunkel-ucd The following updates to search are in this setup:
- 834d4cd If you leave out `q=`, or it's empty then it will return everything.  It's still sorted by score, so we have to revisit that 

- cd3c2ae includes a section aggregating on type,availability and status

```bash
http http://localhost/api/search   q=='' Authorization:"Bearer $jwt" size==10 | jq .aggregations
```

``` json
{
  "availability": {
    "ark:/87287/d7mh2m/keyword/c-ucd-avail/collaborative%20projects": 2,
    "ark:/87287/d7mh2m/keyword/c-ucd-avail/community%20partnerships": 1,
    "ark:/87287/d7mh2m/keyword/c-ucd-avail/industry%20projects": 1,
    "ark:/87287/d7mh2m/keyword/c-ucd-avail/media%20enquiries": 1
  },
  "type": {
    "grant": 91,
    "grant_research": 91,
    "grant_service": 85,
    "expert": 4
  },
  "status": {
    "completed": 86,
    "active": 5
  }
}
```
- 834d4cd you can add `status=[completed,actve]` to your query.  Note that (right now) only grants have a status, so this is also limiting results to grants only.

```bash
http http://localhost/api/search status=completed Authorization:"Bearer $jwt" size==10 | jq .aggregations
```

```json
 http http://localhost/api/search status==completed Authorization:"Bearer $jwt" size==10 | jq .aggregations
{
  "availability": {},
  "type": {
    "grant": 86,
    "grant_research": 86,
    "grant_service": 81
  },
  "status": {
    "completed": 86
  }
}
```

**Comment by qjhart** [comment]
@UcDust , I'm not sure what you were planning with 01a5d97, but I removed those experts-client files again in 8ac5562

**Review by qjhart — APPROVED** [review]

**Pull request merged**